src/protocol/attribution/accumulate_credit.rs

```diff
@@ -176,30 +166,32 @@ impl<'a, F: Field> AccumulateCredit<'a, F> {
         // first, calculate [successor.helper_bit * successor.trigger_bit]
         let mut b = ctx
             .narrow(&Step::HelperBitTimesIsTriggerBit)
-            .multiply(
-                record_id,
-                &successor.report.helper_bit,
-                &successor.report.is_trigger_bit,
-            )
+            .multiply(record_id, &successor.helper_bit, &successor.is_trigger_bit)
             .await?;
 
         // since `stop_bits` is initialized with `[1]`s, we only multiply `stop_bit` in the second and later iterations
         if !first_iteration {
             b = ctx
-                .narrow(&Step::BTimesStopBit)
+                .narrow(&Step::BTimesCurrentStopBit)
                 .multiply(record_id, &b, &current.stop_bit)
                 .await?;
         }
 
-        let credit_future =
-            ctx.narrow(&Step::BTimesSuccessorCredit)
-                .multiply(record_id, &b, &successor.credit);
+        let credit_future = ctx.narrow(&Step::BTimesSuccessorCredit).multiply(
+            record_id,
+            &b,
+            &successor.interaction_bit,
+        );
```

Comment:
I extracted out this interaction pattern logic and generalized it. I changed the field name to `interaction_value` since it could be field values. Otherwise, there's no change to the underlying logic.